core/apiCalls.js

```diff
@@ -176,3 +176,46 @@ export function fetchImageStatusApi(uuid) {
   });
   return p;
 }
+
+export function fetchComposeQueueApi() {
+  const p = new Promise((resolve, reject) => {
+    utils.apiFetch(constants.get_compose_queue)
+    .then(data => {
+      const queue = data.new.concat(data.run);
+      resolve(queue);
+    })
+    .catch(e => {
+      console.log(`Error fetching diff: ${e}`);
+      reject();
+    });
+  });
+  return p;
+}
+
+export function fetchComposeFinishedApi() {
+  const p = new Promise((resolve, reject) => {
+    utils.apiFetch(constants.get_compose_finished)
+    .then(data => {
+      resolve(data.finished);
+    })
+    .catch(e => {
+      console.log(`Error fetching diff: ${e}`);
+      reject();
+    });
+  });
+  return p;
+}
+
+export function fetchComposeFailedApi() {
+  const p = new Promise((resolve, reject) => {
+    utils.apiFetch(constants.get_compose_failed)
+    .then(data => {
+      resolve(data.failed);
+    })
+    .catch(e => {
+      console.log(`Error fetching diff: ${e}`);
```

Comment:
Should this message be more specific for each case?

Reply:
Probably. Any opinions on how should we handle the error?

Reply:
I'm assuming "Error fetching diff" was added for /api/v0/blueprints/diff/.

I'd just have them identify the api call that's being made, like "Error fetching finished composes", etc...

Also, updates were made in #294 regarding the syntax for these. This syntax:
```console.log(`Error fetching diff: ${e}`);```
was updated to this:
`console.log('Error fetching diff', e);`